extensions/reactive-messaging-http/runtime/src/main/java/io/quarkus/reactivemessaging/http/runtime/ReactiveHttpHandlerBean.java

```diff
@@ -0,0 +1,114 @@
+package io.quarkus.reactivemessaging.http.runtime;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import javax.annotation.PostConstruct;
+import javax.inject.Inject;
+import javax.inject.Singleton;
+
+import org.eclipse.microprofile.reactive.messaging.Message;
+import org.jboss.logging.Logger;
+
+import io.quarkus.reactivemessaging.http.runtime.config.HttpStreamConfig;
+import io.quarkus.reactivemessaging.http.runtime.config.ReactiveHttpConfig;
+import io.smallrye.mutiny.Multi;
+import io.smallrye.mutiny.subscription.BackPressureStrategy;
+import io.smallrye.mutiny.subscription.MultiEmitter;
+import io.vertx.core.buffer.Buffer;
+import io.vertx.core.http.HttpMethod;
+import io.vertx.ext.web.RoutingContext;
+
+/**
+ * a bean that handles incoming http requests
+ */
+@Singleton
+public class ReactiveHttpHandlerBean {
+
+    private static final Logger log = Logger.getLogger(ReactiveHttpHandlerBean.class);
+
+    @Inject
+    ReactiveHttpConfig config;
+
+    private final Map<String, Bundle<HttpMessage<?>>> httpProcessors = new HashMap<>();
+
+    @PostConstruct
+    void init() {
+        config.getHttpConfigs()
+                .forEach(this::addHttpProcessor);
+    }
+
+    void handle(RoutingContext event) {
+        String path = event.normalisedPath();
+        HttpMethod method = event.request().method();
+        String key = key(path, method);
+        Bundle<HttpMessage<?>> httpProcessorBundle = httpProcessors.get(key);
+        if (httpProcessorBundle != null) {
+            MultiEmitter<? super HttpMessage<?>> emitter = httpProcessorBundle.emitter;
+            StrictQueueSizeGuard guard = httpProcessorBundle.guard;
+            if (guard.prepareToEmit()) {
```

Comment:
when I tried to marry buffering with checking the number of requested items, I was getting Integer.MAX_VALUE from `requested()`. Can play with it a bit more, maybe I missed something...

Reply:
Did you have an `onOverflow()`?
Actually, depending on your pipeline, it may request Long.MAX_VALUE. You would need to test with a processor / subscriber emitting requests.

Reply:
I'm 5 nines sure that I tested both, with onOverflow and without

Reply:
And I was wrong. If I have `onOverflow` (even with limited buffer), I'm always getting `MAX_VALUE` from `requested()`.
Otherwise, it's 1 or 0.

If I understand correctly, without `onOverflow`, if I'm able to get into `onSubscribe` and request `bufferSize` of elements, it should more or less work (the difference being it won't be as strict as the guard solution because multiple threads can check the if with the same value and emit messages).

Reply:
We can keep the guard.